### PR TITLE
[CBRD-24893] Get stmt type from dblink query.

### PR DESCRIPTION
**Pull request by airnet73**
http://jira.cubrid.org/browse/CBRD-24893

Purpose
The gateway does not have a query parser, so it is difficult to know the stmt type.
So, dblink server added stmt type to query comment.
The gateway can know the stmt type through comment of the query.

Implementation
N/A

Remarks
N/A

### CODE REVIEW

File: src/broker/cas_execute.c [hunk 1]
```diff
@@ -11753,4 +11765,36 @@ ux_cgw_free_stmt (T_SRV_HANDLE * srv_handle)
...
+
+	      ut_trim (type_name);
+	      ut_tolower (type_name);
+
+	      stmt_type = get_stmt_type (type_name);
+	    }
+	}
+    }
+
+  if (stmt_type == CUBRID_STMT_NONE)
```
Comment (kisoo-han) [inline src/broker/cas_execute.c]: get_stmt_type () returns **CUBRID_MAX_STMT_TYPE** if there is no string matching the input argument. Is it intended that it does not return CUBRID_STMT_NONE?

File: src/broker/cas_execute.c [hunk 2]
```diff
@@ -11753,4 +11765,36 @@ ux_cgw_free_stmt (T_SRV_HANDLE * srv_handle)
 {
   cgw_free_stmt (srv_handle);
 }
+
+static char
+ux_cgw_get_stmt_type (char *stmt)
+{
+  char stmt_type = CUBRID_STMT_NONE;
+  const char *comment_start = strstr (stmt, "/*");
+  if (comment_start)
```
Comment (kisoo-han) [inline src/broker/cas_execute.c]: If hints other than the STMT type hint could be sent to the gateway, hint parsing will have to be repeated. If STMT type hint is only one, its ok.

File: src/broker/cas_cgw.c [hunk 3]
```diff
@@ -1770,6 +1770,12 @@ cgw_get_num_cols (SQLHSTMT hstmt, SQLSMALLINT * num_cols)
 
   SQL_CHK_ERR (hstmt, SQL_HANDLE_STMT, err_code = SQLNumResultCols (hstmt, num_cols));
 
+  if (*num_cols == 0)
+    {
+      SQL_CHK_ERR (hstmt, SQL_HANDLE_STMT, err_code = SQLExecute (hstmt));
+      SQL_CHK_ERR (hstmt, SQL_HANDLE_STMT, err_code = SQLNumResultCols (hstmt, num_cols));
+    }
+
   return (int) err_code;
```
Comment (kisoo-han) [inline src/broker/cas_cgw.c:1779]: Maybe you combined two different issues in one jira issue and a pull request. Next time, obey the rule that a jira issue should only cover one issue at a time.

**Review by kisoo-han — APPROVED** [review]

File: src/broker/cas_execute.c (hunk 1 above)
Comment (airnet73) [reply src/broker/cas_execute.c]: Exception handling will be added when CUBRID_MAX_STMT_TYPE is returned.

File: src/broker/cas_execute.c (hunk 2 above)
Comment (airnet73) [reply src/broker/cas_execute.c]: The current format has a hint at the beginning of sql, so only the first hint was searched. ( /* DBLINK SELECT */ select * from t1; )
I will consider the case of multiple hints.

File: src/broker/cas_cgw.c (hunk 3 above)
Comment (airnet73) [reply src/broker/cas_cgw.c:1779]: I will definitely separate them from next time.

**Review by kisoo-han — APPROVED** [review]

**Pull request merged**